src/components/Upload/Upload.js

```diff
@@ -31,16 +32,17 @@ export default class Upload extends Component {
       private: !isPublic,
     };
     // store in both firebase & pg
-    actions.createPuzzle(puzzle, (pid) => {
+    await actions.createPuzzle(puzzle, (pid) => {
```

Comment:
this await seems unnecessary, though feel free to leave it in since i'll be deleting `actions` entirely at some point since it's no longer used 😅